sailpoint.sgnl.yaml

```diff
@@ -0,0 +1,1427 @@
+        externalId: $.application.displayName
+        type: String
+      - name: alertDate
+        externalId: alertDate
+        type: DateTime
+      - name: lastProcessed
+        externalId: lastProcessed
+        type: DateTime
+      - name: nativeId
+        externalId: nativeId
+        type: String
+      - name: targetId
+        externalId: targetId
+        type: String
+      - name: targetType
+        externalId: targetType
+        type: String
+      - name: targetDisplayName
+        externalId: targetDisplayName
+        type: String
+
+      # Meta
+      - name: resourceType
+        externalId: $.meta.resourceType
+        type: String
+      - name: created
+        externalId: $.meta.created
+        type: DateTime
+      - name: lastModified
+        externalId: $.meta.lastModified
+        type: DateTime
+      - name: version
+        externalId: $.meta.version
+        type: String
+      - name: location
+        externalId: $.meta.location
+        type: String
+
+  AlertAttribute:
+    parent: Alerts
+    displayName: SailpointAlertAttributes
+    externalId: attributes
+    description: Entity representing the attibutes associated with the Sailpoint Identity IQ Alert.
+    pageSize: 100
+    pagesOrderedById: false
+    attributes:
+      - name: key
+        externalId: key
+        type: String
+        indexed: true
+        uniqueId: true
+      - name: value
+        externalId: value
+        type: String
+
+  AlertAction:
+    parent: Alerts
+    displayName: SailpointAlertActions
+    externalId: actions
+    description: Entity representing the actions associated with the Sailpoint Identity IQ Alert.
+    pageSize: 100
+    pagesOrderedById: false
+    attributes:
+      - name: resultId
+        externalId: resultId
+        type: String
+        indexed: true
+        uniqueId: true
+      - name: type
+        externalId: type
+        type: String
+      - name: alertDefinitionName
+        externalId: alertDefinitionName
+        type: String
+
+  AlertActionResult:
+    parent: actions
+    displayName: SailpointAlertActionResultNotifications
+    externalId: result
+    description: Entity representing the result associated with the Sailpoint Identity IQ Alert Action.
+    pageSize: 100
+    pagesOrderedById: false
+    attributes:
+      - name: workflowName
+        externalId: workflowName
+        type: String
+        indexed: true
+        uniqueId: true
+      - name: name
+        externalId: name
+        type: String
+
+  AlertActionResultNotification:
+    parent: result
+    displayName: SailpointAlertActionResultNotifications
+    externalId: notifications
+    description: Entity representing the notifications associated with the Sailpoint Identity IQ Alert Action Result.
+    pageSize: 100
+    pagesOrderedById: false
+    attributes:
+      - name: emailAddresses
+        externalId: emailAddresses
+        type: String
+        indexed: true
+        uniqueId: true
```

Comment:
The Sailpoint docs indicate that this is a list:
https://developer.sailpoint.com/iiq/api/get-alerts/

> A list of email addresses to send an Alert action notification.

Ingestion will fail for this attribute.
And I don't think that you'll want to use a list as `uniqueId`.
Setting `list: true` won't solve the problem.
Sorry, I don't see a good solution for this.

Reply:
You're correct that the documentation specifies it is a list. However, looking at the sample response under `application+scim`, we see that it is a collection of comma separated email addresses represented as a string.

```
    "notifications": [
        {
            "name": "lowLevelAlertNotification",
            "displayName": "lowLevelAlert result notification",
            "emailAddresses": "amanda.ross@demoexample.com, alice.ford@demoexample.com"
        }
    ]
```

Setting `list: true` would result in an ingestion failure. Although this qualifies as a valid `uniqueId`, I agree that indexing this attribute would not be fully correct.